qdrant_client/qdrant_remote.py

```diff
@@ -892,6 +892,166 @@ def recommend_groups(
+                grpc.DiscoverPoints(
+                    collection_name=collection_name,
+                    target=target,
+                    context=context,
+                    filter=query_filter,
+                    limit=limit,
+                    offset=offset,
+                    with_vectors=with_vectors,
+                    with_payload=with_payload,
+                    params=search_params,
+                    using=using,
+                    lookup_from=lookup_from,
+                    read_consistency=consistency,
+                    timeout=timeout,
+                ),
+                timeout=timeout if timeout is not None else self._timeout,
+            )
+
+            return [GrpcToRest.convert_scored_point(hit) for hit in res.result]
+        else:
+            target = (
+                GrpcToRest.convert_target_vector(target)
+                if target is not None and isinstance(target, grpc.TargetVector)
+                else target
+            )
+
+            context = [
+                GrpcToRest.convert_context_example_pair(pair)
+                if isinstance(pair, grpc.ContextExamplePair)
+                else pair
+                for pair in context
+            ]
+
+            if isinstance(query_filter, grpc.Filter):
+                query_filter = GrpcToRest.convert_filter(model=query_filter)
+
+            if isinstance(search_params, grpc.SearchParams):
+                search_params = GrpcToRest.convert_search_params(search_params)
+
+            if isinstance(with_payload, grpc.WithPayloadSelector):
+                with_payload = GrpcToRest.convert_with_payload_selector(with_payload)
+
+            if isinstance(lookup_from, grpc.LookupLocation):
+                lookup_from = GrpcToRest.convert_lookup_location(lookup_from)
+
+            result = self.openapi_client.points_api.discover_points(
+                collection_name=collection_name,
+                consistency=consistency,
+                timeout=timeout,
+                discover_request=models.DiscoverRequest(
+                    target=target,
+                    context=context,
+                    filter=query_filter,
+                    params=search_params,
+                    limit=limit,
+                    offset=offset,
+                    with_payload=with_payload,
+                    with_vector=with_vectors,
+                    lookup_from=lookup_from,
+                    using=using,
+                ),
+            ).result
+            assert result is not None, "Discover points API returned None"
+            return result
+
+    def discover_batch(
+        self,
+        collection_name: str,
+        requests: Sequence[types.DiscoverRequest],
+        consistency: Optional[types.ReadConsistency] = None,
+        timeout: Optional[int] = None,
+        **kwargs: Any,
+    ) -> List[List[types.ScoredPoint]]:
+        if self._prefer_grpc:
+            requests = [
```

Comment:
fails if context or target is None

```python
    client = qdrant_client.QdrantClient(prefer_grpc=True)
    http_requests = [models.DiscoverRequest(
        target=1,
        limit=10,
    )]
    result = client.discover_batch(
        collection_name=collection_name,
        requests=http_requests
    )
    print(result)

    http_requests = [models.DiscoverRequest(
        context=[models.ContextExamplePair(positive=positive_id, negative=negative_id)],
        limit=10,
    )]
    result = client.discover_batch(
        collection_name=collection_name,
        requests=http_requests
    )
    print(result)
```

Reply:
Does not work if offset is not set explicitly (in `models.DiscoverRequest` we can avoid setting offset explicitly):

```python
client = QdrantClient(prefer_grpc=True)
target = q_grpc.TargetVector(single=q_grpc.VectorExample(id=q_grpc.PointId(num=positive_id)))
grpc_requests = [q_grpc.DiscoverPoints(
    collection_name=collection_name,
    target=target,
    context=[grpc_context_example_pair],
    limit=10,
    # offset=0,
)]
result = client.discover_batch(
    collection_name=collection_name,
    requests=grpc_requests
)
print(result)
```

Reply:
This offset bug is present also in `search` and `recommend` endpoints, due core not allowing `"offset"=null`. 
I have already raised a PR (qdrant/qdrant#3082) to fix this in core, so we should be able to update for those fixes once they get merged. Though they can be part of a different qdrant-client PR.